automation/gtm_automation.js

```diff
@@ -0,0 +1,496 @@
+  const gtmURL = document.getElementById('gtm-url').value;
+  const config = new Map();
+  config.set('parent', gtmURL.substring(gtmURL.indexOf('accounts')));
+  config.set('ga4ID', document.getElementById('ga4-id').value);
+  config.set('metrics', new Map(Object.entries({
+    ttfb: document.getElementById('cwv-TTFB').checked.toString(),
+    fcp: document.getElementById('cwv-FCP').checked.toString(),
+    lcp: document.getElementById('cwv-LCP').checked.toString(),
+    cls: document.getElementById('cwv-CLS').checked.toString(),
+    fid: document.getElementById('cwv-FID').checked.toString(),
+    inp: document.getElementById('cwv-INP').checked.toString(),
+  })));
+  config.set('useUnpkg', document.getElementById('use-unpkg').checked);
+  config.set('libraryUrl', document.getElementById('library-url').value);
+  config.set('attributionBuild', document.getElementById('use-attribution').checked);
+  config.set('gtmEventName', document.getElementById('event-name').value);
+
+  // get the template content from the user. The file chooser is a mandiatory
+  // field.
+  const templateFile = document.getElementById('template-file').files[0];
+  const templateReader = new FileReader();
+  templateReader.addEventListener('load', () => {
+    createTemplate(config, templateReader.result)
+  });
+  templateReader.readAsText(templateFile);
+}
+
+/**
+ * Creates the template used to measure the Core web Vital Metrics in the
+ * provided workspace.
+ *
+ * If the template is successfully created, the config and the new template ID
+ * is passed to createTag() to continue the deployment.
+ *
+ * @param config {Map<string, *>} The configuration map.
+ * @param content {string} The content of the template.
+ */
+function createTemplate(config, content) {
+  tokenClient.callback = (resp) => {
+    if (resp.error !== undefined) {
+      throw (resp);
+    }
+    gapi.client.tagmanager.accounts.containers.workspaces.templates.create({
+      parent: config.get('parent'),
+      templateData: content,
+    })
+      .then((gtmResp) => {
+        addSuccessMessage('Created Web Vitals Template');
+        const templateID = gtmResp.result.templateId;
+        createTag(config, templateID);
+      })
+      .catch((err) => {
+        addErrorMessage('Error creating template:', JSON.stringify(err));
+      });
+
+  };
+  // This is needed for the first API call to show the consent dialog.
+  // Afterwards the API call can be made directly.
+  if (gapi.client.getToken() == null) {
+    tokenClient.requestAccessToken({ prompt: 'consent' });
+  } else {
+    tokenClient.requestAccessToken({ prompt: '' });
+  }
+}
+
+/**
+ * Creates a tag using the previously installed template to measure CWVs.
+ *
+ * If the tag is successfully created, `createEventTrigger` is called.
+ *
+ * @param config {Map<string, *>} The configuration map.
+ * @param templateID {string} The ID the template was assigned when it was
+ * created.
+ */
+function createTag(config, templateID) {
```

Comment:
nit: Should be templateId
https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
function createTag(config, templateId) {
```

Reply:
Done